client.go

```diff
@@ -0,0 +1,107 @@
+package loki
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/pkg/errors"
+	"go.k6.io/k6/lib"
+	"go.k6.io/k6/lib/netext/httpext"
+)
+
+const (
+	ContentTypeProtobuf   = "application/x-protobuf"
+	ContentTypeJSON       = "application/json"
+	ContentEncodingSnappy = "snappy"
+	ContentEncodingGzip   = "gzip"
+)
+
+type Client struct {
+	client *http.Client
+	logger log.Logger
+	cfg    *Config
+}
+
+type Config struct {
+	URL        url.URL
+	UserAgent  string
+	Timeout    time.Duration
+	TenantID   string
+	EncodeJSON bool
+}
+
+func (c *Client) Push(ctx context.Context) (httpext.Response, error) {
+	state := lib.GetState(ctx)
+	if state == nil {
+		return *httpext.NewResponse(ctx), errors.New("state is nil")
+	}
+
+	entries := generateEntries(c.cfg.TenantID, 100)
```

Comment:
I think we will want to be able to configure more:
- Cardinality generated.
- Format json/apache
- batch size in bytes.